src/libraries/System.Private.CoreLib/src/System/Reflection/FieldAccessor.cs

```diff
@@ -0,0 +1,512 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Reflection
+{
+    internal sealed class FieldAccessor
+    {
+        private readonly RtFieldInfo _fieldInfo;
+        private IntPtr _addressOrOffset;
+        private unsafe MethodTable* _methodTable;
+        private FieldAccessorType _fieldAccessType;
+
+        internal FieldAccessor(FieldInfo fieldInfo)
+        {
+            _fieldInfo = (RtFieldInfo)fieldInfo;
+
+            if (_fieldInfo.m_declaringType.ContainsGenericParameters ||
+                _fieldInfo.m_declaringType.IsNullableOfT)
+            {
+                _fieldAccessType = FieldAccessorType.NoInvoke;
+            }
+            else
+            {
+                _fieldAccessType = FieldAccessorType.SlowPathUntilClassInitialized;
+            }
+        }
+
+        private void Initialize()
+        {
+            Debug.Assert(_fieldInfo.m_declaringType != null);
+
+            if (!RuntimeFieldHandle.IsFastPathSupported(_fieldInfo))
+            {
+                // Currently this is true for [ThreadStatic] cases, for fields added from EnC, and for fields on unloadable types.
+                _fieldAccessType = FieldAccessorType.SlowPath;
+                return;
+            }
+
+            unsafe
+            {
+                _methodTable = (MethodTable*)_fieldInfo.FieldType.TypeHandle.Value;
```

Comment:
This is not going to be MethodTable* for pointer or functiton pointer fields.

Reply:
Change to only set _methodTable when it's used (primitives and value types).